scripts/mirna_extension.py

```diff
@@ -79,159 +123,132 @@ def extend_mirnas(
+and precursor annotations will be written to separate GFF3 files.
+
+Constraints:
+The script was written for GFF3 files obtained from miRBase (http://www.mirbase.org/).
+The following assumptions are made:
+- The input GFF3 file contains miRNA annotations for a single species.
+- The input GFF3 file contains features of type "miRNA_primary_transcript" (referred to
+  as "precursors") and "miRNA" (referred to as "mature miRNAs").
+- The input GFF3 file contains a Name attribute for each precursor.
+- Each precursor contains one or more mature miRNAs.
+- Each mature miRNA is a child of exactly one precursor and is completely within the
+  boundaries of the precursor.
+"""
     parser = argparse.ArgumentParser(
-        description="Script to extend miRNAs start and end coordinates."
+        description=description,
+        formatter_class=argparse.RawDescriptionHelpFormatter,
     )
     parser.add_argument(
         "-v",
         "--version",
         action="version",
-        version="%(prog)s 1.0",
-        help="Show program's version number and exit",
+        version="%(prog)s 1.1.0",
+        help="show program's version number and exit",
     )
     parser.add_argument(
         "input",
-        help="Path to the GFF3 annotation file. If not provided, the input \
-             will be read from the standard input.",
-        type=Path,
-    )
-    parser.add_argument(
-        "--outdir",
-        help="Path to the output directory. Default: %(default)s.",
-        default=Path.cwd(),
+        help="path to the GFF3 annotation file",
         type=Path,
     )
     parser.add_argument(
         "-e",
         "--extension",
-        help="Number of nucleotides to extend the coordinates. Default: \
-                %(default)d.",
+        help=(
+            "number of nucleotides by which to extend the miRNA boundaries; default:"
+            " %(default)d"
+        ),
         default=6,
         type=int,
+        metavar="INT",
     )
     parser.add_argument(
         "--chr",
-        help="Path to the tabulated file containing the chromosome and its \
-            length in basepairs. If not provided, the length will be set to \
-            the biggest coordinate of the last miRNA primary transcript.",
+        help=(
+            "path to a tabulated file containing the chromosome names and their"
+            " lengths, in basepairs; if not provided, the sequence lengths will be"
+            " set to the most distal 3' boundary of the miRNA precursors annotated"
+            " for each chromosome in the input GFF3 file; default: %(default)s"
+        ),
         default=None,
         type=Path,
+        metavar="PATH",
+    )
+    parser.add_argument(
+        "--outdir",
+        help="path to the output directory; default: %(default)s",
+        type=Path,
+        default=Path.cwd(),
+        metavar="PATH",
     )
-
     return parser
 
 
-def main(arguments) -> None:
+def main(args) -> None:
```

Comment:
I believe the usage of "arguments" instead of "args" was due a static code analysis failing if doing so.

Reply:
Here's the error from pylint:

```
mirna_extension.py:247:9: W0621: Redefining name 'args' from outer scope (line 268) (redefined-outer-name)
```

I can change `args` in the last two lines instead of in the main function

Reply:
Ah, yes, makes sense. Yes, I think it would be better to use `arguments` in the scope where you use it less.